### PR TITLE
Finished Lab1

### PR DESCRIPTION
**Pull request by jjoel1630**

### CODE REVIEW

File: joel/lab1/part-1/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.15)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
```
Comment (RCMast3r) [inline joel/lab1/part-1/CMakeLists.txt:4]: prefer to use setting a specific target's compile options: https://cmake.org/cmake/help/latest/command/target_compile_options.html

File: joel/lab1/part-1/src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using PIS = pair<int, string>;
+using VI = vector<int>;
+using VS = vector<string>;
+using VC = vector<char>;
+using VB = vector<bool>;
+using VII = vector<vector<int>>;
+using MII = map<int, int>;
+using MSI = map<string, int>;
+using MSB = map<string, bool>;
+using MCI = map<char, int>;
```
Comment (RCMast3r) [inline joel/lab1/part-1/src/main.cpp:33]: uh what is all this? please do not include random stuff that you dont use.

File: joel/lab1/part-1/src/main.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+//Global Variables:
+void setIO(string name = "") { // name is nonempty for USACO file I/O
+	ios_base::sync_with_stdio(0); cin.tie(0); // see Fast Input & Output
+	if(sz(name)){
+		freopen((name+".in").c_str(), "r", stdin); // see Input & Output
+		freopen((name+".out").c_str(), "w", stdout);
+	}
+}
```
Comment (RCMast3r) [inline joel/lab1/part-1/src/main.cpp:42]: why is this here? please do not include this

File: joel/lab1/part-1/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+project(Onboarding VERSION 1.0
+                  DESCRIPTION "Onboarding"
+                  LANGUAGES CXX)
+
+add_library(hello STATIC hello_world.cpp hello_world.hpp)
```
Comment (RCMast3r) [inline joel/lab1/part-1/CMakeLists.txt:10]: part 1 shouldnt have a library

**Review by RCMast3r — CHANGES_REQUESTED** [review]

**Comment by jjoel1630** [comment]
Fixed changes & added lab2 part 1

**Comment by RCMast3r** [comment]
Next time please re-request my review so I get notified to come and re-check your code

File: joel/lab2/part1/flake.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    easy_cmake.url = "github:RCMast3r/easy_cmake";
+  };
+  outputs = { self, nixpkgs, utils, easy_cmake }:
+    let
+      vn_lib_overlay = final: prev: {
+        vn_lib = final.callPackage ./default.nix { };
+      };
+      py_vn_lib_overlay = final: prev: {
+        py_vn_lib = final.callPackage ./python.nix { };
+      };
```
Comment (RCMast3r) [inline joel/lab2/part1/flake.nix:16]: Please change names and remove what you don't use

File: joel/lab2/part1/flake.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    in
+    {
+      overlays.default = nixpkgs.lib.composeManyExtensions my_overlays;
+
+      packages.aarch64-darwin =
+        rec {
+          vn_lib = pkgs.vn_lib;
+          py_vn_lib = pkgs.py_vn_lib;
+          default = vn_lib;
+        };
```
Comment (RCMast3r) [inline joel/lab2/part1/flake.nix:31]: Ditto about names and removing what you don't use

File: joel/lab2/part1/flake.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+      devShells.aarch64-darwin.default =
+        pkgs.mkShell rec {
+          # Update the name to something that suites your project.
+          name = "nix-devshell";
+          packages = with pkgs; [
+            # Development Tools
+            cmake
+            vn_lib
```
Comment (RCMast3r) [inline joel/lab2/part1/flake.nix:40]: Naming

File: joel/lab2/part1/result
Comment (RCMast3r) [inline joel/lab2/part1/result:1]: Please do not push your result symlink

File: joel/lab2/part1/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.15)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
```
[the remark this answers is not in the document]
Comment (jjoel1630) [reply joel/lab2/part1/CMakeLists.txt:4]: done

File: joel/lab2/part1/default.nix [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{ stdenv, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "easy_cmake";
```
[the remark this answers is not in the document]
Comment (jjoel1630) [reply joel/lab2/part1/default.nix:4]: done

File: joel/lab2/part1/flake.nix [hunk 10]
```diff
@@ -0,0 +1,55 @@
+{
+  description = "fsae onboarding c++ library";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
+    utils.url = "github:numtide/flake-utils";
+    easy_cmake.url = "github:RCMast3r/easy_cmake";
```
[the remark this answers is not in the document]
Comment (jjoel1630) [reply joel/lab2/part1/flake.nix:7]: removed

File: joel/lab2/part1/flake.nix [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  };
+  outputs = { self, nixpkgs, utils, easy_cmake }:
```
[the remark this answers is not in the document]
Comment (jjoel1630) [reply joel/lab2/part1/flake.nix:9]: removed

File: joel/lab2/part1/result
Comment (jjoel1630) [reply joel/lab2/part1/result:1]: deleted

File: joel/lab2/part1/flake.nix (hunk 5 above)
Comment (jjoel1630) [reply joel/lab2/part1/flake.nix:13]: done

File: joel/.gitignore
Comment (RCMast3r) [inline joel/.gitignore:1]: please use just the parent .gitignore

**Review by RCMast3r — CHANGES_REQUESTED** [review]

**Review by RCMast3r — APPROVED** [review]
